internal/cmd/rsl/record/record.go

```diff
@@ -12,20 +14,32 @@ type options struct{}
 
 func (o *options) AddFlags(_ *cobra.Command) {}
 
-func (o *options) Run(_ *cobra.Command, args []string) error {
+func (o *options) Run(cmd *cobra.Command, args []string) error {
 	repo, err := repository.LoadRepository()
 	if err != nil {
 		return err
 	}
 
-	return repo.RecordRSLEntryForReference(args[0], true)
+	if err := repo.RecordRSLEntryForReference(args[0], true); err != nil {
+		return err
+	}
+
+	if err := repo.VerifyRef(cmd.Context(), args[0], false); err != nil {
+		if err := repo.InvalidateLatestRSLEntry(); err != nil {
+			return err
```

Comment:
could you wrap this error so we know why / when this happened?

Reply:
Fixed